### PR TITLE
DoctorAndPositionControllers

### PR DESCRIPTION
**Pull request by BohdanYosypenko**
Add controllers and services for position and Doctor entity

### CODE REVIEW

File: VetClinic.API.Tests/Controllers/PositionControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        private readonly Mock<IPositionService> positionServiceMock;
+        private readonly Mock<IMapper> mapper;
+        private readonly PositionController positionController;
+        public PositionControllerTests()
+        {
+            var fixture = new Fixture();
+            positionServiceMock = fixture.Freeze<Mock<IPositionService>>();
+            mapper = fixture.Freeze<Mock<IMapper>>();
+            positionController = new PositionController(positionServiceMock.Object, mapper.Object);
+
```
Comment (Serafinium) [inline VetClinic.API.Tests/Controllers/PositionControllerTests.cs:27]: Please remove extra spaces

File: VetClinic.API.Tests/Controllers/PositionControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            var positionController1 = new PositionController(positionServiceMock1.Object, mapper1.Object);
+
+            // Act
+            var actualResult = await positionController1.Show(position.Id);
+
+            // Assert                          
+            Assert.True(actualResult.Result is NotFoundResult);
+            positionServiceMock1.Verify(m => m.GetPositionAsync(position.Id), Times.Once);
+
+        }
```
Comment (Serafinium) [inline VetClinic.API.Tests/Controllers/PositionControllerTests.cs:99]: Please remove extra spaces

File: VetClinic.API.Tests/Controllers/PositionControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            // Assert    
+            var result = actualResult.Result as OkObjectResult;
+            Assert.Equal(positionDTO, result.Value);
+            Assert.True(actualResult.Result is OkObjectResult);
+            positionServiceMock.Verify(m => m.GetPositionAsync(position.Id), Times.Once);
+        }
+
+
+        [Fact]
+        public async Task GetById_PositionId_ReturnsNotFound__withNewMock()
```
Comment (Serafinium) [inline VetClinic.API.Tests/Controllers/PositionControllerTests.cs:77]: Correct the name. Use names that are closer to the language of the business ( __withNewMock - what does it means)

File: VetClinic.API/DTO/Doctor/CreateDoctorDto.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public class CreateDoctorDto
+    {       
+        public string Education { get; set; }
+        public string Biography { get; set; }
+        public string Experience { get; set; }
+        public string Photo { get; set; }
+        public int PositionId { get; set; }
+
+
```
Comment (Serafinium) [inline VetClinic.API/DTO/Doctor/CreateDoctorDto.cs:13]: Please remove extra spaces

File: VetClinic.API/DTO/Doctor/ReadDoctorDto.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    public class ReadDoctorDto
+    {       
+        public int Id { get; set; }
+        public string Education { get; set; }
+        public string Biography { get; set; }
+        public string Experience { get; set; }
+        public string Photo { get; set; }
+        public int PositionId { get; set; }
+
+
```
Comment (Serafinium) [inline VetClinic.API/DTO/Doctor/ReadDoctorDto.cs:14]: Please remove extra spaces

File: VetClinic.API/Validators/Doctor/CreateDoctorDtoValidator .cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿using FluentValidation;
+using VetClinic.API.DTO.Doctor;
+
+namespace VetClinic.API.Validators.Doctor
+{
+    public class CreateDoctorDtoValidator : AbstractValidator<CreateDoctorDto>
+    {
+        public CreateDoctorDtoValidator()
+        {
+
```
Comment (Serafinium) [inline VetClinic.API/Validators/Doctor/CreateDoctorDtoValidator .cs:10]: Please add TO DO if you plan to implement this part of code later

File: VetClinic.API/Validators/Doctor/ReadDoctorDtoValidator.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿using FluentValidation;
+using VetClinic.API.DTO.Doctor;
+
+namespace VetClinic.API.Validators.Doctor
+{
+    public class ReadDoctorDtoValidator : AbstractValidator<ReadDoctorDto>
+    {
+        public ReadDoctorDtoValidator()
+        {
+
```
Comment (Serafinium) [inline VetClinic.API/Validators/Doctor/ReadDoctorDtoValidator.cs:10]: Same as above

File: VetClinic.API/Validators/User/CreateUserDtoValidator.cs [hunk 8]
```diff
@@ -7,20 +8,26 @@ public class CreateUserDtoValidator : AbstractValidator<CreateUserDto>
     {
         public CreateUserDtoValidator()
         {
-            RuleFor(user => user.UserName).NotEmpty().WithMessage("Нікнейм не може бути порожнім");
```
Comment (Serafinium) [inline VetClinic.API/Validators/User/CreateUserDtoValidator.cs:10]: Let's use English explanation. Please use English meassages

File: VetClinic.API/Validators/User/UserDtoValidator.cs [hunk 9]
```diff
@@ -7,17 +8,22 @@ public class UserDtoValidator : AbstractValidator<UserDto>
     {
         public UserDtoValidator()
         {
-            RuleFor(user => user.UserName).NotEmpty().WithMessage("Нікнейм не може бути порожнім");
```
Comment (Serafinium) [inline VetClinic.API/Validators/User/UserDtoValidator.cs:10]: Let's use English explanation. Please use English meassages

File: VetClinic.BLL.Tests/AutoMoqDataAttribute.cs [hunk 10]
```diff
@@ -0,0 +1,26 @@
+﻿using AutoFixture;
+using AutoFixture.AutoMoq;
+using AutoFixture.Xunit2;
+using System.Linq;
+
+namespace VetClinic.API.Tests
+{
+    internal class AutoMoqDataAttribute : AutoDataAttribute
+    {
+
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/AutoMoqDataAttribute.cs:10]: Please remove extra spaces

File: VetClinic.BLL.Tests/Services/DoctorServiceTest.cs [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+            var _doctorService = new DoctorService(_repositoryMock.Object, _userServiceMock.Object, _roleManagerMock.Object);
+
+            // Act
+            var actual = await _doctorService.GetDoctorAsync();
+
+            // Assert
+            Assert.Equal(doctors.Count, actual.Count);
+            _repositoryMock.Verify(m => m.DoctorRepository.GetAsync(null, It.IsAny<Func<IQueryable<Doctor>, IIncludableQueryable<Doctor, object>>>(), null, false), Times.Once);
+
+        }
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/Services/DoctorServiceTest.cs:40]: Please remove extra spaces

File: VetClinic.BLL.Tests/Services/DoctorServiceTest.cs [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+            var _doctorService = new DoctorService(_repositoryMock.Object, _userServiceMock.Object, _roleManagerMock.Object);
+
+            // Act    
+            var actual= await _doctorService.AddDoctorAsync(doctor, user);
+
+            // Assert            
+            Assert.Equal(doctor, actual);            
+            
+
+        }
```
Comment (Serafinium) [inline VetClinic.BLL.Tests/Services/DoctorServiceTest.cs:117]: Please remove extra spaces (In all place) :))

File: VetClinic.BLL/Services/Realizations/DoctorService.cs [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+        public async Task<bool> UpdateDoctorAsync(Doctor inputDoctor, User inputUser, int doctorId)
+        {
+            var role = await _roleManager.FindByNameAsync("doctor");
+            if (role == null)
+            {
+                await _roleManager.CreateAsync(new IdentityRole() { Name = "doctor" });
+                role = await _roleManager.FindByNameAsync("doctor");
+            }
+            
+            var doctor= await _repositoryWrapper.DoctorRepository.GetFirstOrDefaultAsync(
```
Comment (Serafinium) [inline VetClinic.BLL/Services/Realizations/DoctorService.cs:92]: Please format the code

File: VetClinic.API.Tests/Controllers/PositionControllerTests.cs (hunk 1 above)
Comment (BohdanYosypenko) [reply VetClinic.API.Tests/Controllers/PositionControllerTests.cs:27]: I remove all unnecessary spaces.

File: VetClinic.API/Validators/Doctor/CreateDoctorDtoValidator .cs (hunk 6 above)
Comment (BohdanYosypenko) [reply VetClinic.API/Validators/Doctor/CreateDoctorDtoValidator .cs]: Wrote comment for this part.

**Pull request merged**